apps/emqx_durable_storage/src/emqx_ds_cache.hrl

```diff
@@ -0,0 +1,36 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+-ifndef(EMQX_DS_CACHE_HRL).
+-define(EMQX_DS_CACHE_HRL, true).
+
+-define(CACHE_KEY(LASTKEY), {LASTKEY}).
+-define(EOS_KEY, []).
+-define(EOS_MARKER, []).
+
+-record(cache_entry, {
+    key :: ?CACHE_KEY(emqx_ds:message_key()) | ?EOS_KEY,
+    seqno :: seqno(),
+    inserted_at :: timestamp(),
+    message :: emqx_types:message() | ?EOS_MARKER
+}).
+-type cache_entry() :: #cache_entry{}.
+
+-type seqno() :: non_neg_integer().
```

Comment:
Nit: It's better to avoid defining types in the headers, since they'll end up duplicated in each module using the hrl.